ruma-api-macros/src/api/request.rs

```diff
@@ -209,26 +212,13 @@ impl TryFrom<RawRequest> for Request {
                     field_kind = Some(match meta {
                         Meta::Word(ident) => {
                             match &ident.to_string()[..] {
-                                s @ "body" | s @ "raw_body" => {
-                                    if let Some(f) = &newtype_body_field {
-                                        let mut error = syn::Error::new_spanned(
-                                            field,
-                                            "There can only be one newtype body field",
-                                        );
-                                        error.combine(syn::Error::new_spanned(
-                                            f,
-                                            "Previous newtype body field",
-                                        ));
-                                        return Err(error);
-                                    }
-
-                                    newtype_body_field = Some(field.clone());
-                                    match s {
-                                        "body" => RequestFieldKind::NewtypeBody,
-                                        "raw_body" => RequestFieldKind::NewtypeRawBody,
-                                        _ => unreachable!(),
-                                    }
-                                }
+                                attr @ "body" | attr @ "raw_body" => util::req_res_meta_word(
+                                    attr,
+                                    &field,
+                                    &mut newtype_body_field,
+                                    || RequestFieldKind::NewtypeBody,
+                                    || RequestFieldKind::NewtypeRawBody,
```

Comment:
Is there any way to make this function generic but a bit less funky, or is the param `fn() -> T` not terrible?